src/composeSupergraph.ts

```diff
@@ -0,0 +1,101 @@
+import execa from "execa";
+import debug from "debug";
+import { healthcheckRouter, healthcheck } from "./utils/client";
+import { logWithTimestamp, writeableDebugStream } from "./utils/logging";
+import { resolve } from "path";
+import { readFile, writeFile } from "fs/promises";
+import { createWriteStream } from "fs";
+
+const roverDebug = debug("rover");
+
+/**
+ * Composes supergraph using rover dev command and introspection.
+ * 
+ * @param productsUrl products schema URL 
+ */
+export async function composeDevSupergraph(productsUrl: string, productsSchema?: string): Promise<Boolean> {
+    logWithTimestamp("composing supergraph...");
+    roverDebug(`\n***********************\nComposing supergraph...\n***********************\n\n`);
+
+    // composing supergraph
+    //   cannot use supergraph.config.js as we need to run rover dev sequentially for each subgraph
+    //   this is a workaround to https://github.com/apollographql/rover/issues/1258
+    // TODO cleanup once multiple subgraphs could be specified with single rover dev command
+    await composeDevSubgraph("products", productsUrl, productsSchema);
+    await composeDevSubgraph("users", "http://localhost:4002", resolve(__dirname, "..", "subgraphs", "users", "users.graphql"));
+    await composeDevSubgraph("inventory", "http://localhost:4003", resolve(__dirname, "..", "subgraphs", "users", "users.graphql"));
+
+    const started = await healthcheckRouter();
+    if (started) {
+        roverDebug(`\n***********************\nSupergraph composed...\n***********************\n\n`);
+        return true;
+    } else {
+        return false;
+    }
+}
+
+async function composeDevSubgraph(subgraphName: string, subgraphUrl: string, schemaFile?: string) {
+    roverDebug(`Composing supergraph - loading ${subgraphName} schema`);
+
+    const started = await healthcheck(subgraphName, subgraphUrl);
```

Comment:
Maybe try/catch here as well (and in other healthcheck call sites)

Reply:
`healtcheck` already handles errors